### PR TITLE
rust/sip: register parser for tcp v12

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3351

Describe changes:
- Rebase

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1538
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by glongo** [comment]
I've just set the correct `SV_BRANCH`, please restart the checks if possible.

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 181 | 195 | 107.73% | 

Pipeline 17018

**Comment by ct0br0** [comment]
Re-ran TLPW2, uptime sometimes matches baseline and other times is a bit up.

File: rust/src/sip/sip.rs [hunk 1]
```diff
@@ -147,20 +148,75 @@ impl SIPState {
...
+        let input = stream_slice.as_slice();
+        if input.is_empty() {
+            return AppLayerResult::ok();
+        }
+        let _pdu = Frame::new(
+            flow,
+            &stream_slice,
+            input,
+            input.len() as i64,
+            SIPFrameType::Pdu as u8,
```
Comment (catenacyber) [inline rust/src/sip/sip.rs:163]: This frame does not look right :
A TCP sent here may have 0 or multiple SIP PDUs

This should be done after `sip_parse_request` returned `Ok`

Cf https://github.com/OISF/suricata/pull/10048 (MQTT current implementation is wrong)

**Review by catenacyber — CHANGES_REQUESTED** [review]
Frames on tcp should support multiple PDUs in one call to `parse_request_tcp` (same for response)

**Comment by glongo** [comment]
New PR: https://github.com/OISF/suricata/pull/10058

**Pull request closed without merging**